packages/alert-dialog/src/AlertDialog.ts

```diff
@@ -0,0 +1,209 @@
+     * `variant` attribute is removed when not matching one of the above.
+     *
+     * @param {String} variant
+     */
+    @property({ type: String, reflect: true })
+    public set variant(variant: AlertDialogVariants) {
+        if (variant === this.variant) {
+            return;
+        }
+        const oldValue = this.variant;
+        if (alertDialogVariants.includes(variant)) {
+            this.setAttribute('variant', variant);
+            this._variant = variant;
+        } else {
+            this.removeAttribute('variant');
+            this._variant = '';
+        }
+        this.requestUpdate('variant', oldValue);
+    }
+
+    public get variant(): AlertDialogVariants {
+        return this._variant;
+    }
+
+    private _variant: AlertDialogVariants = '';
+
+    @property({ attribute: 'cancel-label' })
+    public cancelLabel = '';
+
+    @property({ attribute: 'confirm-label' })
+    public confirmLabel = '';
+
+    @property({ attribute: 'secondary-label' })
+    public secondaryLabel = '';
+
+    @property()
+    public headline = '';
+
+    private renderIcon(variant: string): TemplateResult {
+        switch (variant) {
+            case 'warning':
+            case 'error':
+                return html`
+                    <sp-icon-alert class="icon"></sp-icon-alert>
+                `;
+
+            default:
+                return html``;
+        }
+    }
+
+    protected override get dialog(): Dialog {
+        return this.shadowRoot.querySelector('sp-dialog') as Dialog;
+    }
+
+    private clickSecondary(): void {
+        this.dispatchEvent(
+            new Event('secondary', {
+                bubbles: true,
+            })
+        );
+    }
+
+    private clickCancel(): void {
+        this.dispatchEvent(
+            new Event('cancel', {
+                bubbles: true,
+            })
+        );
+    }
+
+    private clickConfirm(): void {
+        this.dispatchEvent(
+            new Event('confirm', {
+                bubbles: true,
+            })
+        );
+    }
+
+    private getButtonVariantAndTreatment(variant: string): [string, string] {
+        let buttonVariant = '';
+        let buttonTreatment = '';
+
+        switch (variant) {
+            case 'confirmation':
+                buttonVariant = 'accent';
+                break;
+            case 'destructive':
+                buttonVariant = 'negative';
+                break;
+            default:
+                buttonVariant = 'primary';
+                buttonTreatment = 'outline';
+                break;
+        }
+
+        return [buttonVariant, buttonTreatment];
+    }
+
+    protected override renderDialog(): TemplateResult {
+        const [buttonVariant, buttonTreatment] =
+            this.getButtonVariantAndTreatment(this.variant);
+        return html`
+            <div class="alertDialog">
+                <sp-dialog>
```

Comment:
I am still concerned with how we are using <dialog /> with in alert dialog, They should not rely on each other and it is also causing the CSS for dialog to be applied to alert-dialog when it shouldn't. You can see in the screen shot that  .grid  from Dialog is being applied. 

![Screenshot 2023-08-14 at 4 21 16 PM](https://github.com/adobe/spectrum-web-components/assets/63808889/a21fdc4d-7935-463f-a61e-86568ec4bfa2)

Reply:
Yes @jenndiaz This is because I have extended the DialogBase in AlertDialog. Is there any specific reasons that Alert Dialog CSS attributes for Divider is different from Dialog? Can we consider DialogBase as the foundation for dialog and built extended dialogs like Alert on top of it?

Reply:
So previously I was re-using the dialog component to pull most of the UI into the alert dialog. To answer your question, I have created the UI in a new base alert component now so that it can be extended in future if any UI / CSS changes comes in only for alert dialog.